### DISCUSSION TITLE
It is possible to "`ls`" my online OneDrive folder?

### DISCUSSION THREAD

**Discussion by kalyanoliveira** [abraunegg/onedrive#2749] [Q&A] [answered]
Basically, I want to `ls` my online OneDrive folder. I want to see the names of the files that I currently have in my online OneDrive folder, and then use `--single-directory` to download only the things that I deem necessary at any time. 

Is this kind of behavior possible? I don't want to have to download everything into my computer (hundreds of GBs if counting everything).

In Windows, this behavior is very trivial. Simply open up the default file explorer, and navigate to wherever the OneDrive folder is. There, all of your files will be listed as if they were already on your computer. Except, of course, they're not: when you actually click to open a file, the Windows OneDrive client will then (and only then) download it.

I don't necessarily want (or need) this exact behavior, but having the ability to `ls` things from the terminal would be very convenient. I just want to not have to use OneDrive in my browser, really.

**Comment by kalyanoliveira** [comment]
After looking into [the on-demand-related issue of this repository](https://github.com/abraunegg/onedrive/issues/757), I see that [onedriver](https://github.com/jstaf/onedriver) exists. This is not necessarily the solution that I want, though.

Just listing file names, for me, would be enough.

**Comment by abraunegg** [comment, marked as answer]
@kalyanoliveira 
> Is this kind of behavior possible?

Unfortunatly this is not possible. Please raise a feature request.
